base/reflection.jl

```diff
@@ -348,17 +346,41 @@ function raise_match_failure(name::Symbol, @nospecialize(tt))
     error("$name: unanalyzable call given $sig_str")
 end
 
+"""
+    default_compiler_module()
+
+In order to facilitate working on the compiler, we'd like to make it easy for
+our users to switch all the ordinary compiler-based reflection to a new copy of the
+compiler, if they `dev` and load a new version of the compiler. However, it would
+be confusing if the result of reflection functions changed, simply because some
+downstream package happens to load a new copy of `Compiler`. Thus, we look for
+a special rendezvous binding `var"#_interactive_compiler_reference` in Main.
+This binding is exported by the compiler, so if the user does `using Compiler` to
+load a dev'ed version of the Compiler from `Main`, we will see that and pick up
+the new Compiler, but if the using statement is in some downstream package, we
+will not.
+"""
+function default_compiler_module()
+    # If the user has explicitly loaded a new compiler into Main, prefer that.
+    if isdefined(Core.Main, :var"#_interactive_compiler_reference")
+        return Core.Main.var"#_interactive_compiler_reference"
```

Comment:
This resolves the binding, then returns whether it has a value, so it doesn't quite follow the comment behavior. Should we check instead if it is resolved?

Reply:
`isdefined` is not supposed to resolve:
```
julia> module Foo
       const x = 1
       export x
       end

julia> isdefined(Main, :x)
false

julia> using .Foo

julia> isdefined(Main, :x)
true

julia> Main.x
1
```